### PR TITLE
A bunch of unit tests for Zookeeper

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

**Comment by tsuna** [comment]
Any particular reason why this was closed?

File: src/HBaseClient.java [hunk 1]
```diff
@@ -207,22 +207,25 @@
...
-  private static final byte[] META = new byte[] { '.', 'M', 'E', 'T', 'A', '.' };
-  static final byte[] INFO = new byte[] { 'i', 'n', 'f', 'o' };
-  private static final byte[] REGIONINFO = new byte[] { 'r', 'e', 'g', 'i', 'o', 'n', 'i', 'n', 'f', 'o' };
-  private static final byte[] SERVER = new byte[] { 's', 'e', 'r', 'v', 'e', 'r' };
+  protected static final byte[] ROOT = new byte[] { '-', 'R', 'O', 'O', 'T', '-' };
+  protected static final byte[] ROOT_REGION = new byte[] { '-', 'R', 'O', 'O', 'T', '-', ',', ',', '0' };
+  protected static final byte[] META = new byte[] { '.', 'M', 'E', 'T', 'A', '.' };
+  protected static final byte[] INFO = new byte[] { 'i', 'n', 'f', 'o' };
+  protected static final byte[] REGIONINFO = new byte[] { 'r', 'e', 'g', 'i', 'o', 'n', 'i', 'n', 'f', 'o' };
+  protected static final byte[] SERVER = new byte[] { 's', 'e', 'r', 'v', 'e', 'r' };
```
Comment (tsuna) [inline src/HBaseClient.java]: Why are all those becoming protected?

File: src/HBaseClient.java [hunk 2]
```diff
@@ -207,22 +207,25 @@
...
-  private static final byte[] META_REGION_NAME =
+  protected static final byte[] META_REGION_NAME =
     new byte[] { 'h', 'b', 'a', 's', 'e', ':', 'm', 'e', 't', 'a', ',', ',', '1' };
   /** New for HBase 0.95 and up: the region info for META is fixed.  */
-  private static final RegionInfo META_REGION =
+  protected static final RegionInfo META_REGION =
     new RegionInfo(HBASE96_META, META_REGION_NAME, EMPTY_ARRAY);
 
+  /** How many times to retry an RPC before giving up */
+  protected static int MAX_RETRY_ATTEMPTS = 10;
```
Comment (tsuna) [inline src/HBaseClient.java]: Shouldn't this be final?

**Comment by tsuna** [comment]
Since the `test` package is no more, and the test code now lives in the same package as AsyncHBase itself, more things could be made package-private so they can be accessed from the new tests, but I don't believe we should make all this internal stuff `protected`.

**Comment by manolama** [comment]
ah good point I do need to do that.

File: src/HBaseClient.java (hunk 2 above)
Comment (manolama) [inline src/HBaseClient.java]: It actually went away with the config stuff I added

**Comment by tsuna** [comment]
Cool :)

**Pull request closed without merging**